test/unit/util/http.js

```diff
@@ -91,6 +91,21 @@ describe('util/http', () => {
     });
   });
 
+  describe('cookie setter', () => {
+    const { withCookie } = http;
+    const mockRequest = () => ({ set: function(key, value) { this[key] = value; } });
+
+    it('should ultimately return the result', () => {
+      withCookie()()(42)(null, mockRequest()).should.equal(42);
```

Comment:
no response is ever formulated. this test exercises the middleware filter function directly, and so none of that plumbing is ever entered.

Reply:
It makes sense to me that the mock has a `set()` method and nothing else, since that's all that `withCookie()` hits. However, I think that `withCookie()` calls `set()` on the response, not the request: it doesn't use the request at all. I think that means that `mockRequest()` is actually being used as a mock response: after `null` is specified for the first parameter, `mockRequest()` is specified for the second, and that second parameter is `response`.